requests/sessions.py

```diff
@@ -728,7 +728,7 @@ def get_adapter(self, url):
                 return adapter
 
         # Nothing matches :-/
-        raise InvalidSchema("No connection adapters were found for '%s'" % url)
+        raise InvalidSchema("No connection adapters were found for {!r}".format(url))
```

Comment:
What **!r** stands for? Python 2.7 supporting .format as well as you tell. Its ok but I couldn't get first one.

Reply:
`{!r}` in `.format()` is the equivalent to `%r` in  `%`.

Reply:
The original line has 's' flag, why you used 'r'. I don't have enough knowledge about format flags

Reply:
`url` should always be a string. So far `url = 'https://example.com/`, `"No connection adapters were found for '{!s}'".format(url)` is the same as `"No connection adapters were found for {!r}".format(url)`, as both output `"No connection adapters were found for 'https://example.com/'`,  for the raw form of a string includes its quotes. `{!r}` or `repr()` tends to be faster than `{!s}` or `str()`.